### PR TITLE
Malte lig 1112 docker pass relevantfilenames to api

### PR DESCRIPTION
**Pull request by MalteEbner**
## Description
- make`ApiWorkflowClient.download_raw_samples()`, `ApiWorkflowClient.download_raw_predictions()`, `ApiWorkflowClient.download_raw_metadata()` use the same base function handling the cursor
- make all 3 functions support `relevant_filnenames_file_name` argument
-  PR in docker: https://github.com/lightly-ai/lightly-core/pull/1578

### CODE REVIEW

File: tests/api_workflow/test_api_workflow_datasources.py [hunk 1]
```diff
@@ -50,6 +50,23 @@ def test_download_raw_samples_equal_to_download_all_raw_new_samples(self):
...
+        for method in [
+            self.api_workflow_client.download_raw_samples,
+            self.api_workflow_client.download_raw_metadata
+        ]:
+            for relevant_filenames_path in [None, "", "relevant_filenames.txt"]:
+                with self.subTest(
+                        relevant_filenames_path=relevant_filenames_path,
+                        method=method
+                ):
+                    samples = self.api_workflow_client.download_raw_samples(
```
Comment (japrescott) [inline tests/api_workflow/test_api_workflow_datasources.py:64]: shouldn't you be using the method you are passing in? 🤔

**Review by japrescott — COMMENTED** [review]
some questions but the abstraction looks awesome :)

File: tests/api_workflow/mocked_api_workflow_client.py [hunk 2]
```diff
@@ -403,8 +403,20 @@ def get_datasource_processed_until_timestamp_by_dataset_id(
...
+            dataset_id,
+            cursor: str = None,
+            _from: int = None,
+            to: int = None,
+            relevant_filenames_file_name: str = -1
     ) -> DatasourceRawSamplesData:
+        if relevant_filenames_file_name == -1:
+            samples = self._samples[dataset_id]
+        elif isinstance(relevant_filenames_file_name, str) and len(relevant_filenames_file_name) > 0:
+            samples = self._samples[dataset_id][::2]
```
[the remark this answers is not in the document]
Comment (MalteEbner) [reply tests/api_workflow/mocked_api_workflow_client.py:416]: I did not want to re-implement the full logic in the API, which would need to
- create a relevant filenames file given the path
- read the filenames from it
- subset the samples given the filenames.

This should be mocked good enough.

File: tests/api_workflow/test_api_workflow_datasources.py (hunk 1 above)
Comment (MalteEbner) [reply tests/api_workflow/test_api_workflow_datasources.py:64]: Good point, this is an error.

File: tests/api_workflow/mocked_api_workflow_client.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (japrescott) [reply tests/api_workflow/mocked_api_workflow_client.py:417]: all good, was just trying to understand what it does

File: lightly/api/api_workflow_datasources.py [hunk 3]
```diff
@@ -12,8 +11,50 @@
 
 class _DatasourcesMixin:
 
+    def _download_raw_files(
```
[the remark this answers is not in the document]
Comment (MalteEbner) [reply lightly/api/api_workflow_datasources.py:14]: Thank you :)

**Review by japrescott — APPROVED** [review]

**Pull request merged**